policy/resolver.go

```diff
@@ -1034,41 +1034,39 @@ func (cache *policyResolverCache) addCheckJobVariants(ctx context.Context, query
 }
 
 func (cache *policyResolverCache) addDataQueryJob(ctx context.Context, query *explorer.Mquery, ownerJob *ReportingJob) {
-	if len(query.Variants) != 0 {
-		err := cache.addDataQueryVariants(ctx, query, ownerJob)
-		if err != nil {
-			log.Error().Err(err).Str("queryMrn", query.Mrn).Msg("failed to add data query variants")
-		}
-		return
-	}
-
-	uuid := cache.global.relativeChecksum(query.Checksum)
+	uuid := cache.global.relativeChecksum(query.Mrn)
 	queryJob := cache.global.reportingJobsByUUID[uuid]
 
-	// note: the ReportingJob is only a placeholder and is replaced by individual query LLX checksum ReportingJobs
 	if queryJob == nil {
 		queryJob = &ReportingJob{
-			Uuid:       cache.global.relativeChecksum(query.Checksum),
+			Uuid:       uuid,
 			QrId:       query.Mrn,
 			ChildJobs:  map[string]*explorer.Impact{},
 			Datapoints: map[string]bool{},
 			Type:       ReportingJob_DATA_QUERY,
-			// FIXME: DEPRECATED, remove in v10.0 vv
-			DeprecatedV8IsData: true,
 		}
-		cache.global.reportingJobsByUUID[queryJob.Uuid] = queryJob
+		cache.global.codeIdToMrn[query.CodeId] = append(cache.global.codeIdToMrn[query.CodeId], query.Mrn)
+		cache.global.reportingJobsByUUID[uuid] = queryJob
 		cache.global.reportingJobsByMsum[query.Checksum] = append(cache.global.reportingJobsByMsum[query.Checksum], queryJob)
 		cache.childJobsByMrn[query.Mrn] = append(cache.childJobsByMrn[query.Mrn], queryJob)
 	}
 
 	// local aspects for the resolved policy
 	queryJob.Notify = append(queryJob.Notify, ownerJob.Uuid)
-
-	ownerJob.Datapoints[queryJob.Uuid] = true
-	// we set a placeholder for the execution query, just to indicate it will be added
-	cache.global.executionQueries[query.Checksum] = nil
-	cache.global.dataQueries[query.Checksum] = struct{}{}
-	cache.global.queriesByMsum[query.Checksum] = query
+	if ownerJob.ChildJobs[queryJob.Uuid] == nil {
+		ownerJob.ChildJobs[queryJob.Uuid] = query.Impact
```

Comment:
From what I understood, we do not set `Impact`. So, this could be `nil`. At other places in the code, we have these checks:
```
if impact == nil {
	impact = &explorer.Impact{}
}
```
Should we add this here, too?

Reply:
I have copied this over from the `addCheckJob` func, I haven't seen that elsewhere